src/CycloneDX.Core/Models/AlgorithmProperties.cs

```diff
@@ -0,0 +1,357 @@
+        public PaddingScheme Padding_XML
+        {
+            get { return Padding.Value; }
+            set { Padding = value; }
+        }        
+        [Browsable(false), EditorBrowsable(EditorBrowsableState.Never)]
+        public bool ShouldSerializePadding_XML() => Padding.HasValue;
+        #endregion Padding
+
+        [XmlIgnore]
+        [ProtoMember(9)]
+        public List<CryptoFunction> CryptoFunctions { get; set; }
+
+
+        [XmlElement("cryptoFunctions")]
+        [JsonIgnore]
+        public CryptoFunctionCollection CryptoFunctionSerializable_XML
+        {
+            get { return new CryptoFunctionCollection { CryptoFunctions = CryptoFunctions }; }
+            set { CryptoFunctions = new List<CryptoFunction>(value.CryptoFunctions); }
+        }
+
+        [XmlElement("classicalSecurityLevel")]
+        [ProtoMember(10)]
+        public int ClassicalSecurityLevel { get; set; }
+        [XmlElement("nistQuantumSecurityLevel")]
+        [ProtoMember(11)]
+        public int NistQuantumSecurityLevel { get; set; }
+        
+
+    }
+
+    public class CryptoFunctionCollection
+    {
+        [XmlElement("cryptoFunction")]
+        public List<CryptoFunction> CryptoFunctions { get; set; }
+    }
+
+
+    public enum PaddingScheme
+    {       
+        Null,
+        [XmlEnum("pkcs5")]
+        PKCS5,
+        [XmlEnum("pkcs7")]
+        PKCS7,
+        [XmlEnum("pkcs1v15")]
+        PKCS1v15,
+        [XmlEnum("oaep")]
+        OAEP,
+        [XmlEnum("raw")]
+        Raw,
+        [XmlEnum("other")]
+        Other,
+        [XmlEnum("unknown")]
+        Unknown
+
+
+    }
+
+    public enum AlgorithmMode
+    {
+        Null,
+        [XmlEnum("cbc")]
+        Cbc,
+        [XmlEnum("ecb")]
+        Ecb,
+        [XmlEnum("ccm")]
+        Ccm,
+        [XmlEnum("gcm")]
+        Gcm,
+        [XmlEnum("cfb")]
+        Cfb,
+        [XmlEnum("ofb")]
+        Ofb,
+        [XmlEnum("ctr")]
+        Ctr,
+        [XmlEnum("other")]
+        Other,
+        [XmlEnum("unknown")]
+        Unknown
+    }
+
+    public enum CertificationLevel
+    {
+        Null,
+        [XmlEnum("none")]
+        None,
+        [XmlEnum("fips140-1-l1")]
+        FIPS140_1_L1,
+        [XmlEnum("fips140-1-l2")]
+        FIPS140_1_L2,
+        [XmlEnum("fips140-1-l3")]
+        FIPS140_1_L3,
+        [XmlEnum("fips140-1-l4")]
+        FIPS140_1_L4,
+        [XmlEnum("fips140-2-l1")]
+        FIPS140_2_L1,
+        [XmlEnum("fips140-2-l2")]
+        FIPS140_2_L2,
+        [XmlEnum("fips140-2-l3")]
+        FIPS140_2_L3,
+        [XmlEnum("fips140-2-l4")]
+        FIPS140_2_L4,
+        [XmlEnum("fips140-3-l1")]
+        FIPS140_3_L1,
+        [XmlEnum("fips140-3-l2")]
+        FIPS140_3_L2,
+        [XmlEnum("fips140-3-l3")]
+        FIPS140_3_L3,
+        [XmlEnum("fips140-3-l4")]
+        FIPS140_3_L4,
+        [XmlEnum("cc-eal1")]
+        CC_EAL1,
+        [XmlEnum("cc-eal1+")]
+        CC_EAL1plus,
```

Comment:
Will this work correctly? "+" => plus?

Reply:
There are a couple of more instances in this enum.

Reply:
@mtsfoni https://github.com/CycloneDX/cyclonedx-dotnet-library/pull/333 should fix this.